File: AA_15-2/cf_474_B.py
#codeforces div 2 - 474B

def findWorm(find):
	begin = 0
	end = len(intervals) - 1
	mid = (begin + (end - begin)) / 2
	
	if intervals[mid][0] <= find  and find <= intervals[mid][1]: return mid + 1
	
	while begin <= end:
		mid = begin + ((end - begin) / 2)
		
		if find < intervals[mid][0]: end = mid - 1
		elif find > intervals[mid][1]: begin = mid + 1
		else: return mid + 1
	
	return -1
	
	
p = int(raw_input())
np = map(int, raw_input().split())

intervals = []
qtd = 0
for i in range(p):
	if i == 0: qtd = 0
	else: qtd = intervals[-1][1]
	
	pair = (qtd + 1, qtd + np[i])
	intervals.append(pair)
	
q = int(raw_input())
queries = map(int, raw_input().split())
ans = []

for i in range(q):
	a = findWorm(queries[i])
	ans.append(a)

for i in ans:
	print i

File: online_judge_solutions/cf_474_B.cpp
//codeforces div2 - 474 B

#include <bits/stdc++.h>

using namespace std;
typedef long long int ll;

vector<ll> np;
vector<pair<ll, ll> > intervals;

ll findWorm(ll find, ll p) {
	ll begin = 0, end = p-1, mid = (begin + (end - begin)) / 2;
	if(intervals.at(mid).first <= find && intervals.at(mid).second >= find) return mid + 1;
	
	while(begin <= end){
		mid = begin + ((end - begin) / 2);
		
		if(intervals.at(mid).second < find) begin = mid + 1;
		else if(intervals.at(mid).first > find) end = mid -1;
		else return mid + 1;
	}
	
	return -1;
}

int main(){
	ll p;
	cin >> p;
	
	for(ll i = 0; i < p; i++) {
		ll kp;
		cin >> kp;
		np.push_back(kp);
	}
	
	for(ll i = 0; i < p; i++) {
		ll qtd;
		if(i == 0) qtd = 0;
		else qtd = intervals.at(intervals.size()-1).second;
		
		pair<ll, ll> delimitators;
		delimitators.first = qtd + 1;
		delimitators.second = qtd + np.at(i);
		intervals.push_back(delimitators);
	}
	
	ll q;
	cin >> q;
	vector<ll> queries;
	
	for(ll i = 0; i < q; i++) {
		ll qi;
		cin >> qi;
		queries.push_back(qi);
	}
	
	for(ll i = 0; i < q; i++) {
		cout << findWorm(queries.at(i), p) << endl;
	}
	
	return 0;
}
